File: app/Models/StorySnippet.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class StorySnippet extends Model
{
    use HasFactory;

    protected $fillable = [
        'figure_id',
        'title',
        'subtitle',
        'content',
        'content_blocks',
        'image_path',
        'audio_path',
        'audio_status',
        'audio_error',
        'youtube_url',
    ];

    /**
     * Cast attributes to native types.
     */
    protected $casts = [
        'content_blocks' => 'array',
    ];

    /**
     * The figure this story snippet belongs to.
     */
    public function figure(): BelongsTo
    {
        return $this->belongsTo(Figure::class);
    }
}

File: src/admirable/domain/entities/story_snippet.py
from dataclasses import dataclass
from datetime import datetime

from admirable.domain.entities._audio_capable import AudioCapable
from admirable.domain.value_objects.audio_status import AudioStatus
from admirable.domain.value_objects.content_block import ContentBlock, build_search_text


@dataclass
class StorySnippet(AudioCapable):
    id: int | None
    figure_id: int
    title: str
    subtitle: str | None
    content_blocks: list[ContentBlock]
    search_text: str
    image_path: str | None = None
    audio_path: str | None = None
    audio_status: AudioStatus = AudioStatus.IDLE
    audio_error: str | None = None
    youtube_url: str | None = None
    created_at: datetime | None = None
    updated_at: datetime | None = None

    def rebuild_search_text(self) -> None:
        self.search_text = build_search_text(self.content_blocks)
